[Chow Yung Jie Kenneth] iP

reignnz opened the pull request:
# DukeBot

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))


DukeBot allows you to manage your daily tasks easily yet efficiently. It is: 

- Text-based
- Easy to learn
- Super fast to use

All you need to do is,

1. download it from [here](https://github.com/reignnz/ip/releases/tag/A-Jar)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features: 
- [X] Managing Tasks 
- [X] Managing Deadlines
- [ ] Reminders (coming soon) 

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(DukeGui.class, args);
    }
}
```

Commit:
Add Gradle support

reignnz committed:
Added functionality for Level 1

reignnz committed:
Added functionality for Level 2

reignnz committed:
Added functionality for Level 3

reignnz committed:
Added functionality for Level 4

reignnz committed:
Added test cases for A-TextUiTesting

reignnz committed:
Added Error Handling for Level-5

reignnz committed:
Added functionality for Level 6

reignnz committed:
Stored Duke Commands into a single package

reignnz committed:
Add functionality for Level-7. Allows DukeBot to save and load tasks into a local database.

reignnz committed:
Add functionality for Level-8. Allows DukeBot to accept date and times.

reignnz committed:
Merge branch 'branch-Level-7'

reignnz committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/DukeBot/Deadline.java

reignnz committed:
Refactor the code to extract out closely related code as classes.

Organize the classes into suitable java packages.

reignnz committed:
Fix gap in Deadline toString method

reignnz committed:
Add JUnit Tests.

reignnz committed:
Create JAR file

reignnz committed:
Add JavaDoc

reignnz committed:
Add functionality for finding Tasks with the specified keyword (Level-9)

reignnz committed:
Refactor code to follow Coding Standards

reignnz committed:
Merge branch 'branch-A-JavaDoc'

reignnz committed:
Merge branch 'branch-A-CodingStandard'

reignnz committed:
Merge branch 'branch-Level-9'

reignnz committed:
Resolve conflicts of merger from A-CodingStandard, A-JavaDoc and Level-9

reignnz committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

reignnz committed:
Merge branch 'branch-A-Gradle'

junjieteoh commented on src/main/java/Commands/DukeCommand.java:
```diff
@@ -0,0 +1,25 @@
+
+public class DukeCommand {
+    public String commandBody;
+
```
I think it may be better to leave class attribute private.

junjieteoh commented on src/main/java/Commands/DukeCommand.java:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    public void execute(TaskList task, Ui ui, Storage storage) throws DukeException, IOException {
+
```
Will it be better to remove this method since it is empty?

junjieteoh commented on src/main/java/Commands/FindCommand.java:
```diff
@@ -0,0 +1,33 @@
+            }
+        }
+
+        if (arrList.getSize() == 0 ){
```
There is an extra space here.
```suggestion
        if (arrList.getSize() == 0){
```

junjieteoh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,60 @@
+import util.Storage;
+import util.Ui;
+
+import java.io.*;
```
I think it is better to import what is needed instead of importing everything.

junjieteoh commented on src/main/java/Tasks/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+ */
+public class Deadline extends Task {
+
+    public LocalDateTime deadline;
```
I think it is better to leave this as private.

junjieteoh commented on src/main/java/Tasks/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+     * @return
+     */
+    @Override
+    public String dBText() {
```
Will it be better to name it as dbText ?

junjieteoh commented on src/main/java/Tasks/Task.java:
```diff
@@ -0,0 +1,64 @@
+ * Task is a class that contains the details of a task
+ */
+public class Task {
+    private boolean completed;
```
I think it will be nicer to rename this as isComplete

junjieteoh commented on src/main/java/Tasks/TaskList.java:
```diff
@@ -0,0 +1,40 @@
+ * TaskList is a class that deals with the management and storage of tasks
+ */
+public class TaskList {
+    private ArrayList<Task> arrayList;
```
I feel that `tasks` might be a better variable name here.

junjieteoh commented on src/main/java/util/Storage.java:
```diff
@@ -0,0 +1,68 @@
+        // Task | Completeness (1 = complete, 0 = not yet) | description | date by
+        Scanner s = new Scanner(f);
+        while (s.hasNextLine()) {
+            String taskDetails = s.nextLine();
```
I feel that `taskDetail `might be a better name here.

junjieteoh commented on src/main/java/Commands/ByeCommand.java:
```diff
@@ -0,0 +1,27 @@
+package Commands;
```
It is a better convention for package should be named in all lowercase. This is the same across some other package.

junjieteoh reviewed (commented):
LGTM, almost. Just a few bit of minor error in naming of variables and some minor coding convention issues. I have commented on some of the ones that I have found, but there might be some more here and there. Overall, I found your code quite nicely written and cleanly organized.

reignnz committed:
Add Graphical User Interace (GUI) to DukeBot

reignnz committed:
Merge branch 'branch-Level-10'

9teMare commented on src/main/java/Commands/DukeCommand.java:
```diff
@@ -0,0 +1,25 @@
+
+import java.io.IOException;
+
+public class DukeCommand {
```
I think a better practice should be making the `DukeCommand` an abstract class, that way I don't think there is a need to return 'Empty' for the default `execute()`, all the children of this class should implement the `execute()` method in there individually.

9teMare commented on src/main/java/Commands/FindCommand.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException, IOException {
+        TaskList arrList = new TaskList();
```
I think better naming convention can be implemented here, instead of `arrList` which looks like `ArrayList`, you could name it as something like `tasksFound`.

9teMare commented on src/main/java/Commands/InvalidCommand.java:
```diff
@@ -0,0 +1,17 @@
+import util.Storage;
+import util.Ui;
+
+public class InvalidCommand extends DukeCommand {
```
Personally, I think if a `Command` is invalid, it should not be considered as a `Command`, therefore no inheritance. A better implementation to me is probably to remove this particular `InvalidCommand`, and directly throw exceptions when parsing fails.

9teMare commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,77 @@
+
+    }
+
+    /**
+     * Runs the application
+     */
+    /*
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+
+        while (!isExit) {
+            try {
+                String input = ui.readCommand();
+                ui.showLine();
+
+                DukeCommand c = Parser.parse(input);
+                c.execute(tasks, ui, storage);
+                isExit = c.isExit();
+
+            } catch (DukeException e) {
+                ui.showErrorMessage(e.getMessage());
+            } catch (IOException e) {
+                e.printStackTrace();
+            } finally {
+                ui.showLine();
+            }
+        }
+    }
+     */
```
Huge block of commented out code, can be removed.

9teMare commented on src/main/java/Exceptions/EmptyDescriptionException.java:
```diff
@@ -0,0 +1,11 @@
+ * EmptyDescriptionException is a class that deals with empty description bodies followed by the
+ * user inputted String
+ */
+public class EmptyDescriptionException extends DukeException{
```
I like how you categorized your exceptions into subclasses🙂.

9teMare commented on src/main/java/util/Parser.java:
```diff
@@ -0,0 +1,140 @@
+package util;
+
+import Commands.*;
```
I think it is better to import classes from the `Commands` package one by one, instead of using `*` even though you indeed need all of them. In fact, that is what being specified by checkStyle.

9teMare commented on src/main/java/util/Parser.java:
```diff
@@ -0,0 +1,140 @@
+     */
+    public static LocalDateTime parseDateTime(String description, String task) throws DukeException {
+
+        String splitDesc[] = new String[0];
```
I think a better naming convention should be `splitedDescription`, because now it sounds like a verb, which by convention is for method names.

9teMare commented on src/main/java/util/Storage.java:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    public ArrayList<Task> load() throws FileNotFoundException {
+        ArrayList<Task> arr = new ArrayList<>();
```
I think a better naming convention should be implemented, try to name `arr` to exactly what it stores, something like `storedTasks` or `tasksFromDb`.

9teMare reviewed (commented):
Overall well done, I like your customization for the GUI🙂!
Just to point out that better naming conventions can be implemented.

reignnz committed:
Fix DukeBot's Introduction Message in the JavaFx GUI

reignnz committed:
Add assertion to various points in the code base

Assertions can assist in the identification of bugs at a specific location efficiently

Identifying critical points in the code and adding assertions in these areas

This allows us to target areas in the code that are most vulnerable and allow
for fast rectification whenever a problem occurs

Refer to the JavaPoint tutorial about assertions
https://www.javatpoint.com/assertion-in-java

reignnz committed:
Refactor code to adhere to the Code Quality guidelines

To increase the understandability of the code.

Changing portions of code that contain long methods, deep nesting, complicated expressions and magic literals.

This allows the user to better understand and follow the purpose of the code written.

Information about code quality can be found here
https://nus-cs2103-ay2122s2.github.io/website/se-book-adapted/chapters/codeQuality.html#code-quality

reignnz committed:
Merge pull request #1 from reignnz/branch-A-Assertions

Add assertion to various points in the code base

reignnz committed:
Merge pull request #2 from reignnz/branch-A-CodeQuality

Refactor code to adhere to the Code Quality guidelines

reignnz committed:
Merge branch 'master' of https://github.com/reignnz/ip

* 'master' of https://github.com/reignnz/ip:
  Refactor code to adhere to the Code Quality guidelines
  Add assertion to various points in the code base

reignnz committed:
Add feature to view tasks on the specified date

reignnz committed:
Update README.md

reignnz committed:
Update README.md

reignnz committed:
Update README.md

reignnz committed:
Add Ui Image

reignnz committed:
Delete Ui.png

reignnz committed:
Add Ui image

reignnz committed:
Refactor code and improve GUI

reignnz committed:
Merge branch 'master' of https://github.com/reignnz/ip

* 'master' of https://github.com/reignnz/ip:
  Add Ui image
  Delete Ui.png
  Add Ui Image
  Update README.md
  Update README.md
  Update README.md

reignnz committed:
Update JUnit Test File

reignnz committed:
Update README.md

reignnz committed:
Merge branch 'master' of https://github.com/reignnz/ip

* 'master' of https://github.com/reignnz/ip:
  Update README.md

reignnz committed:
Update JUnit Test
